sklearn/utils/testing.py

```diff
@@ -372,14 +375,52 @@ def __exit__(self, *exc_info):
     assert_greater = _assert_greater
 
 
+def _sparse_dense_allclose(val1, val2, rtol=1e-7, atol=0):
```

Comment:
If you're going to make a broad use, scipy-oriented, recursive, all-close comparator, I'd rather it takes the form of an `assert_*`, e.g. `assert_scipy_allclose` or `assert_generic_allclose`... It would be nice to just overload the name `assert_allclose` or `assert_almost_equal`, but that may engender confusion.